bash/drive-utils.bash

```diff
@@ -0,0 +1,529 @@
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    printf "%s\n" "${search_response}"
+    return 0
+}
+
+###################################################
+# Search for an existing file on gdrive with write permission.
+# Globals: 2 variables, 2 functions
+#   Variables - API_URL, API_VERSION
+#   Functions - _url_encode, _json_value
+# Arguments: 3
+#   ${1} = file name
+#   ${2} = root dir id of file
+#   ${3} = Access Token
+# Result: print file id else blank
+# Reference:
+#   https://developers.google.com/drive/api/v3/search-files
+###################################################
+_check_existing_file() {
+    [[ $# -lt 3 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare name="${1##*/}" rootdir="${2}" token="${3}"
+    declare query search_response id
+
+    "${EXTRA_LOG:-:}" "justify" "Checking if" " file exists.." "-" 1>&2
+    query="$(_url_encode "name='${name}' and '${rootdir}' in parents and trashed=false and 'me' in writers")"
+
+    search_response="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
+        -H "Authorization: Bearer ${token}" \
+        "${API_URL}/drive/${API_VERSION}/files?q=${query}&fields=files(id,name,mimeType)&supportsAllDrives=true" || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+    "${EXTRA_LOG_CLEAR:-:}" 1 1>&2
+
+    id="$(_json_value id 1 1 <<< "${search_response}")"
+    [[ -n ${id} ]] && printf "%s\n" "${search_response}"
+    return 0
+}
+
+###################################################
+# Create/Check directory in google drive.
+# Globals: 2 variables, 2 functions
+#   Variables - API_URL, API_VERSION
+#   Functions - _url_encode, _json_value
+# Arguments: 3
+#   ${1} = dir name
+#   ${2} = root dir id of given dir
+#   ${3} = Access Token
+# Result: print folder id
+# Reference:
+#   https://developers.google.com/drive/api/v3/folder
+###################################################
+_create_directory() {
+    [[ $# -lt 3 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare dirname="${1##*/}" rootdir="${2}" token="${3}"
+    declare query search_response folder_id
+
+    "${EXTRA_LOG:-:}" "justify" "Creating gdrive dir.." "-" 1>&2
+    query="$(_url_encode "mimeType='application/vnd.google-apps.folder' and name='${dirname}' and trashed=false and '${rootdir}' in parents")"
+
+    search_response="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
+        -H "Authorization: Bearer ${token}" \
+        "${API_URL}/drive/${API_VERSION}/files?q=${query}&fields=files(id)&supportsAllDrives=true" || :)" && "${CURL_PROGRESS_EXTRA_CLEAR:-:}" 1 1>&2
+
+    folder_id="$(printf "%s\n" "${search_response}" | _json_value id 1 1)"
+
+    if [[ -z ${folder_id} ]]; then
+        declare create_folder_post_data create_folder_response
+        create_folder_post_data="{\"mimeType\": \"application/vnd.google-apps.folder\",\"name\": \"${dirname}\",\"parents\": [\"${rootdir}\"]}"
+        create_folder_response="$(curl --compressed ${CURL_PROGRESS_EXTRA} \
```

Comment:
Codacy found an issue: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5865418)